examples/topics/categorical/correlogram.py

```diff
@@ -0,0 +1,73 @@
+''' A categorical plot showing the correlations in mineral content for 214 samples of glass fragments
+obtained during forensic work.
+
+The dataset contains seven variables measuring the amounts of magnesium (Mg),
+calcium (Ca), iron (Fe), potassium (K), sodium (Na), aluminum (Al), and barium (Ba) found in each glass fragment.
+The colored circles represent the correlations between pairs of these variables.
+The magnitude of each correlation is encoded in the size of the circles.
+
+.. bokeh-example-metadata::
+    :sampledata: forensic_glass
+    :apis: bokeh.plotting.figure.scatter
+    :refs: :ref:`ug_topics_categorical_scatters`
+    :keywords: scatter, correlogram
+
+'''
+
+from itertools import combinations
+
+import numpy as np
+import pandas as pd
+
+from bokeh.models import ColorBar, ColumnDataSource, FixedTicker, LinearColorMapper
+from bokeh.plotting import figure, show
+from bokeh.sampledata.forensic_glass import data as df
+from bokeh.transform import transform
+
+elements = ("Mg", "Ca", "Fe", "K", "Na", "Al", "Ba")
+
+pairs = list(combinations(elements, 2))
+x, y = list(zip(*pairs))
+
+correlations = []
+for x, y in pairs:
+    matrix = np.corrcoef(df[x], df[y])
+    correlations.append(matrix[0, 1])
+
+new_df = pd.DataFrame({
+    "oxide_1": [x[0] for x in pairs],
+    "oxide_2": [x[1] for x in pairs],
+    "correlation": correlations,
+    "dot_size": [(1+ 10 * abs(corr)) * 10 for corr in correlations],
```

Comment:
is `correlations` a numpy array? In that case also no need for a list comprehension, can just use array ops:
```
10 * (1+ 10 * abs(correlations)) 
```